src/components/popover-container/popover-container.pw.tsx

```diff
@@ -0,0 +1,683 @@
+
+    await expect(popoverCloseIconElement).toHaveAttribute(
+      "aria-label",
+      testString
+    );
+  });
+
+  test("should render with containerAriaLabel", async ({ mount, page }) => {
+    await mount(<PopoverContainerComponent containerAriaLabel={testString} />);
+
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).toHaveAttribute(
+      "aria-label",
+      testString
+    );
+  });
+
+  test("should render with renderOpenComponent", async ({ mount, page }) => {
+    await mount(<WithRenderOpenButtonComponent />);
+
+    const buttonComponent = getComponent(page, "button");
+    await buttonComponent.getByText("Test");
+
+    await expect(buttonComponent).toBeVisible();
+
+    const filterNewButton = getDataElementByValue(page, "filter_new");
+
+    await expect(filterNewButton).toBeVisible();
+
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).not.toBeVisible();
+  });
+
+  test("should render with renderCloseComponent", async ({ mount, page }) => {
+    await mount(<WithRenderCloseButtonComponent />);
+
+    const getComponentElementButton = getComponent(page, "button");
+    await getComponentElementButton.getByText("Test");
+
+    await expect(getComponentElementButton).toBeVisible();
+
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).toBeVisible();
+  });
+
+  const getBottomValue = (locator: Locator) =>
+    locator.evaluate((element) => element.getBoundingClientRect().bottom);
+
+  const getTopValue = (locator: Locator) =>
+    locator.evaluate((element) => element.getBoundingClientRect().top);
+
+  const getYValue = (locator: Locator) =>
+    locator.evaluate((element) => element.getBoundingClientRect().y);
+
+  ([
+    [true, 102, 178],
+    [false, 126, 202],
+  ] as const).forEach(([coverButton, yAndTopValueMin, bottomValueMin]) => {
+    test(`should render with shouldCoverButton prop set to ${coverButton}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <Box height="330px">
+          <PopoverContainerComponentCoverButton
+            shouldCoverButton={coverButton}
+          />
+        </Box>
+      );
+
+      const popoverContainerContentElement = popoverContainerContent(page);
+      await expect(popoverContainerContentElement).toBeInViewport();
+
+      const positionTop = await getTopValue(popoverContainerContentElement);
+      const positionBottom = await getBottomValue(
+        popoverContainerContentElement
+      );
+      const positionY = await getYValue(popoverContainerContentElement);
+
+      await expect(positionBottom).toBeLessThan(bottomValueMin);
+      await expect(positionTop).toBeLessThan(yAndTopValueMin);
+      await expect(positionY).toBeLessThan(yAndTopValueMin);
+    });
+  });
+
+  ([...keyToTrigger] as const).forEach((key) => {
```

Comment:
```suggestion
  keyToTrigger.forEach((key) => {
```

**suggestion**: various suggestions here, which apply to all the other tests using this array:

- there's absolutely no need to spread it, we can just refer to the array itself
- the `as const` is unnecessary, there's already an `as const` in the definition (I'm not even sure if that's necessary but I haven't checked, and even if it's not I don't think it does any harm)
- not shown here, but as I've said in other PRs, I'd rather rename it `keysToTrigger` because it's an array and has more than one key in it